app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = [
+        Person(person["name"], person["age"])
+        for person in people
+    ]
+
+    for index, person in enumerate(people):
+        if person.get("wife") is not None:
+            setattr(
+                person_list[index],
+                "wife",
+                Person.people[person["wife"]]
+            )
```

Comment:
You could make this code more efficient using walrus operator
```suggestion
        if wife_name := person.get("wife"):
            setattr(
                person_list[index],
                "wife",
                Person.people[wife_name]
            )
```